app/src/main/java/com/duckduckgo/app/browser/BrowserTabViewModel.kt

```diff
@@ -281,6 +285,31 @@ class BrowserTabViewModel(
         skipHome = false
     }
 
+    suspend fun onUserSubmittedAutocomplete(suggestion: AutoCompleteApi.AutoCompleteSuggestion) {
+        onUserSubmittedQuery(suggestion.phrase)
+
+        val hasBookmarks = withContext(Dispatchers.IO) { bookmarksDao.hasBookmarks() }
+        fireAutocompletePixel(suggestion, hasBookmarks)
+    }
+
+    private fun fireAutocompletePixel(suggestion: AutoCompleteApi.AutoCompleteSuggestion, hasBookmarks: Boolean) {
+        val currentViewState = currentAutoCompleteViewState()
+        val params = mapOf(
+            Pixel.PixelParameter.SHOWED_BOOKMARKS to currentViewState.searchResults.hasBookmarks.toString(),
```

Comment:
this doesn't seem to work as is; `SHOWED_BOOKMARKS` is always false. the reason is that `currentAutoCompleteViewState()` has already been updated after the query/URL has been cleared after selecting it.

Reply:
I noticed that problem while doing some final testing yesterday night. This has now changed so it happens before updating the `autoCompleteViewState`